### DISCUSSION TITLE
Upgrade from Version 1 to Version 2

### DISCUSSION THREAD

**Discussion by leifbln** [fabianishere/udm-iptv#37] [Q&A]
Hi all,

 i'm struggling a bit to do an Upgrade from Version 1 to Version 2. 

Do you have a Walkthrough how to proceed Upgrading to Version 2 from the older first Version? 

Since all Commands etc. are not anymore available i'm a bit helpless here  :( 

Thanks!

**Comment by fabianishere** [comment]
I recommend having a look at the installation instructions: https://github.com/fabianishere/udm-iptv#installation
Is there a step there where you get stuck?

**Reply by leifbln** [reply]
Hi Fabian, yes wanted to add it this moment: 

![image](https://user-images.githubusercontent.com/95283373/149540161-e6d0d710-947e-4f24-bc9c-fe50e3817f85.png)

I'm not able to Download the udm-iptv resources.

**Reply by fabianishere** [reply]
Please enter UniFi OS first: `unifi-os shell`

**Reply by leifbln** [reply]
Much worse...

![image](https://user-images.githubusercontent.com/95283373/149540555-86718630-94c2-4c2a-9239-12e5cb10fba3.png)

**Reply by fabianishere** [reply]
What about using `curl -O` instead of `wget`?

**Reply by leifbln** [reply]
![image](https://user-images.githubusercontent.com/95283373/149541128-b36bf033-6391-4412-9978-e2033fcfab10.png)

**Reply by fabianishere** [reply]
This should work: `curl -O -L https://github.com/fabianishere/udm-iptv/releases/download/v2.0.2/udm-iptv_2.0.2_all.deb`

**Reply by leifbln** [reply]
Thanks, that looks good. Do you think the old installation will interfere with the new one? Should I remove anything from the Old Deployment?

**Reply by fabianishere** [reply]
Remove the bootscript at `/mnt/data/on_boot.d/15-iptv.sh` and then remove your device.

**Reply by StreborStrebor** [reply]
@fabianishere I too am not sure what needs to be done to upgrade from 1.x to 2.0.3 

I have UDM with firmware 1.11.0 and I suppose a 1.x version of the script. 
I see the file /mnt/data/on_boot.d/15-iptv.sh. Removing this is not a problem.

But, what do you mean by "and then remove your device"?

**Reply by StreborStrebor** [reply]
I seem to have upgraded fine, by simply removing /mnt/data/on_boot.d/15-iptv.sh
Then running the CURL downloads and finally installing and configuring with the APT command
Finally a reboot, and all is working again.
GREAT!
